src/Tvheadend.cpp

```diff
@@ -2224,6 +2291,31 @@ void CTvheadend::ParseChannelAddOrUpdate(htsmsg_t* msg, bool bAdd)
       /* CAID */
       if (caid == 0)
         htsmsg_get_u32(&f->hmf_msg, "caid", &caid);
+
+      /* Service provider */
+      str = htsmsg_get_str(&f->hmf_msg, "providername");
+      if (str)
+      {
+        const uint32_t uid{utilities::hash_str_uint32(str)};
```

Comment:
Empty provider names should probably be ignored?

Reply:
Yep. Thanks.

Reply:
done